File: course/src/main/java/in/testpress/course/ui/CoursePreviewActivity.java
package in.testpress.course.ui;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import androidx.annotation.Nullable;

import java.util.ArrayList;

import in.testpress.course.R;
import in.testpress.ui.BaseToolBarActivity;

import static in.testpress.course.TestpressCourse.COURSE_IDS;
import static in.testpress.course.TestpressCourse.PRODUCT_SLUG;


public class CoursePreviewActivity extends BaseToolBarActivity {

    private static final String TAG = "CoursePreviewActivity";

    public static Intent createIntent(ArrayList<Integer> courseIds, Context context, String productSlug) {
        Intent intent = new Intent(context, CoursePreviewActivity.class);
        intent.putExtra(COURSE_IDS, courseIds);
        intent.putExtra(PRODUCT_SLUG, productSlug);
        return intent;
    }

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.testpress_container_layout);
        setContentView(R.layout.testpress_container_layout);
        CoursePreviewFragment fragment = new CoursePreviewFragment();
        fragment.setArguments(getIntent().getExtras());
        getSupportFragmentManager().beginTransaction().replace(R.id.fragment_container, fragment)
                .commitAllowingStateLoss();
    }
}

File: course/src/main/java/in/testpress/course/ui/CoursePreviewActivity.kt
package `in`.testpress.course.ui

import android.content.Context
import android.content.Intent
import android.os.Bundle
import `in`.testpress.course.R
import `in`.testpress.course.TestpressCourse
import `in`.testpress.store.TestpressStore
import `in`.testpress.ui.BaseToolBarActivity

class CoursePreviewActivity: BaseToolBarActivity() {
    private var isPurchaseSuccessful = false
    private var continuePurchase = false

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.testpress_container_layout)
        val fragment = CoursePreviewFragment()
        fragment.arguments = intent.extras
        supportFragmentManager.beginTransaction().replace(R.id.fragment_container, fragment)
            .commitAllowingStateLoss()
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)
        if (requestCode == TestpressStore.STORE_REQUEST_CODE && resultCode == RESULT_OK && data != null) {
            isPurchaseSuccessful = true
            continuePurchase = data.getBooleanExtra(TestpressStore.CONTINUE_PURCHASE, false)
            if (!continuePurchase) {
                setResult(RESULT_OK, data)
                finish()
            }
        }
    }

    override fun finish() {
        if (isPurchaseSuccessful) {
            val intent = Intent().apply {
                putExtra(TestpressStore.PAYMENT_SUCCESS, true)
                putExtra(TestpressStore.CONTINUE_PURCHASE, continuePurchase)
            }
            setResult(RESULT_OK, intent)
        }
        super.finish()
    }

    companion object {

        @JvmStatic
        fun createIntent(
            courseIds: ArrayList<Int>,
            context: Context,
            productSlug: String?
        ): Intent {
            return Intent(context, CoursePreviewActivity::class.java).apply {
                putExtra(TestpressCourse.COURSE_IDS, courseIds)
                putExtra(TestpressCourse.PRODUCT_SLUG, productSlug)
            }
        }
    }
}